includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1267,6 +1267,11 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 		if ( $payment_needed ) {
 			$payment_method_details = $intent->get_payment_method_details();
 			$payment_method_type    = $payment_method_details ? $payment_method_details['type'] : null;
+
+			if ( $order->get_meta( 'is_woopay' ) ) {
+				$order->add_meta_data( 'last4', $payment_method_details['card']['last4'], true );
```

Comment:
I think for now the WooPay payment method will always be card payment, but should we add a check that `$payment_method_type` is `'card'` before adding the metadata, just in case?

Reply:
Fixed with 37e3d23d2f6ace956ea1526021e53779c32ad830